diff --git a/dist/index.mjs b/dist/index.mjs
index a34e2816ac9b0e83dd28346a76da8232d7d02aa8..7374ad5edcbb361ad397ed8ee04672a4b9338a71 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -68,11 +68,7 @@ function __awaiter(thisArg, _arguments, P, generator) {
 let otelModulePromise = null;
 const OTEL_PKG = "@opentelemetry/api";
 function loadOtel() {
-	if (otelModulePromise === null) otelModulePromise = import(
-		/* webpackIgnore: true */
-		/* @vite-ignore */
-		OTEL_PKG
-).catch(() => null);
+	if (otelModulePromise === null) otelModulePromise = Promise.resolve(null);
 	return otelModulePromise;
 }
 /**
