diff --git a/src/index.tsx b/src/index.tsx
index 38357aa..278d32f 100644
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -519 +519 @@
-          onTranslationYChange(Math.abs(ty), shouldClose.value);
+          onTranslationYChange(ty, shouldClose.value);
diff --git a/lib/commonjs/index.js b/lib/commonjs/index.js
index 38ff3dc..c02a5d3 100644
--- a/lib/commonjs/index.js
+++ b/lib/commonjs/index.js
@@ -431 +431 @@
-        onTranslationYChange(Math.abs(ty), shouldClose.value);
+        onTranslationYChange(ty, shouldClose.value);
diff --git a/lib/module/index.js b/lib/module/index.js
index 07f595e..c3200c1 100644
--- a/lib/module/index.js
+++ b/lib/module/index.js
@@ -406 +406 @@
-        onTranslationYChange(Math.abs(ty), shouldClose.value);
+        onTranslationYChange(ty, shouldClose.value);
